linera-core/src/chain_worker/state/mod.rs

```diff
@@ -423,6 +424,18 @@ where
     async fn create_network_actions(&self) -> Result<NetworkActions, WorkerError> {
         let mut heights_by_recipient = BTreeMap::<_, BTreeMap<_, _>>::new();
         let mut targets = self.chain.outboxes.indices().await?;
+
+        if let Some(limit) = self.config.network_actions_batch_limit {
+            if limit > targets.len() / 2 {
+                let elements_to_discard = targets.len().saturating_sub(limit);
+                targets.partial_shuffle(&mut thread_rng(), elements_to_discard);
```

Comment:
Why not just use `choose_multiple`? That should be faster, not require the case distinction, and the _order_ of the selected elements doesn't really matter that much. I guess the downside is that it clones the entries, but they are small, I think?